jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ContextProvider.java

```diff
@@ -641,4 +580,308 @@ protected void pathChanged(Path path) throws Exception
+     * a single {@link Properties} instance.
+     * </p>
+     *
+     * @param directory the directory to load environment properties from.
+     * @param env the environment name
+     */
+    private Properties loadEnvironmentProperties(Path directory, String env) throws IOException
+    {
+        Properties props = new Properties();
+        List<Path> envPropertyFiles = new ArrayList<>();
+
+        // Get all environment specific properties files for this environment,
+        // order them according to the lexical ordering of the filenames
+        try (Stream<Path> paths = Files.list(directory))
+        {
+            envPropertyFiles = paths.filter(Files::isRegularFile)
+                .map(directory::relativize)
+                .filter(p ->
+                {
+                    String name = p.getName(0).toString();
+                    if (!name.endsWith(".properties"))
+                        return false;
+                    if (!name.startsWith(env))
+                        return false;
+                    return true;
+                }).sorted().toList();
+        }
+
+        if (LOG.isDebugEnabled())
+            LOG.debug("Environment property files {}", envPropertyFiles);
+
+        // Load each *.properties file
+        for (Path file : envPropertyFiles)
+        {
+            Path resolvedFile = directory.resolve(file);
+            if (Files.exists(resolvedFile))
+            {
+                Properties tmp = new Properties();
+                try (InputStream stream = Files.newInputStream(resolvedFile))
+                {
+                    tmp.load(stream);
+                    //put each property into our substitution pool
+                    tmp.stringPropertyNames().forEach(k -> props.put(k, tmp.getProperty(k)));
+                }
+            }
+        }
+
+        return props;
+    }
+
+    private void loadProperties(Environment environment, InputStream inputStream) throws IOException
+    {
+        Properties props = new Properties();
+        props.load(inputStream);
+        props.stringPropertyNames().forEach((name) ->
+            environment.setAttribute(name, props.getProperty(name)));
+    }
+
+    /**
+     * Builder of a deployment configuration for a specific {@link Environment}.
+     *
+     * <p>
+     * Results in {@link Attributes} for {@link Environment} containing the
+     * deployment configuration (as {@link Deployable} keys) that is applied to all deployable
+     * apps belonging to that {@link Environment}.
+     * </p>
+     */
+    public static class EnvironmentConfig
+    {
+        // Using setters in this class to allow jetty-xml <Set name="" property="">
+        // syntax to skip setting of an environment attribute if property is unset,
+        // allowing the in code values to be same defaults as they are in embedded-jetty.
+
+        private final Environment environment;
```

Comment:
This file and `ScanningAppProvider` use the `_` prefix to class fields, so this inner  class should do the same.